src/wayland/seat/keyboard.rs

```diff
@@ -313,22 +354,24 @@ impl KeyboardHandle {
     /// to be compared against. This includes non-character keysyms, such as XF86 special keys.
     pub fn input<F>(&self, keycode: u32, state: KeyState, serial: Serial, time: u32, filter: F)
     where
-        F: FnOnce(&ModifiersState, Keysym) -> bool,
+        F: FnOnce(&ModifiersState, KeysymHandle<'_>) -> bool,
```

Comment:
This is only slightly related, but looking back at that that API, I wonder if it would make sense to have this closure instead return an enum `{ Forward, Intercept(T) }`, and have the `T` value be returned by the `input()` method as an `Option<T>` ?

This would allow compositors to just return the chosen action and process it afterwards, rather than either processing it from withing the callback or storing it in a captured variable (like we do in anvil).

That's a small QoL improvement I suppose, so maybe it's not worth it. What do you think?

Reply:
I have added it. It removed less lines from anvil then I anticipated, but it is still a better api imo. I am just not 100% sure about the name of the enum.